core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -256,6 +268,14 @@ public void close() throws IOException {
       if (mChannel.isOpen()) {
         mChannel.pipeline().removeLast();
       }
+      if (closeFuture != null) {
```

Comment:
Should this go before the previous statement, otherwise there can be a race between closing the channel and the `mChannel.isOpen` call?